R/flatline_epi_forecaster.R

```diff
@@ -0,0 +1,116 @@
+#' Predict the future with today's value
+#'
+#' This is a simple forecasting model for
+#' [epiprocess::epi_df] data. It uses the most recent observation as the
+#' forcast for any future date, and produces intervals based on the quantiles
+#' of the residuals of such a "flatline" forecast over all available training
+#' data.
+#'
+#' By default, the predictive intervals are computed separately for each
+#' combination of key values (`geo_value` + any additional keys) in the
+#' `epi_data` argument.
+#'
+#' This forecaster is very similar to that used by the
+#' [COVID19ForecastHub](https://covid19forecasthub.org)
+#'
+#' @param epi_data An [epiprocess::epi_df]
+#' @param outcome A scalar character for the column name we wish to predict.
+#' @param args_list A list of dditional arguments as created by the
+#'   [flatline_args_list()] constructor function.
+#'
+#' @return A data frame of point (and optionally interval) forecasts at a single
+#'   ahead (unique horizon) for each unique combination of `key_vars`.
+#' @export
+#'
+#' @examples
+#' jhu <- case_death_rate_subset %>%
+#'   dplyr::filter(time_value >= as.Date("2021-12-01"))
+#'
+#' out <- flatline_epi_forecaster(jhu, "death_rate")
+flatline_epi_forecaster <- function(epi_data,
+                                    outcome,
+                                    args_list = flatline_args_list()) {
+
+  validate_forecaster_inputs(epi_data, outcome, "time_value")
+  keys <- epi_keys(epi_data)
+  ek <- keys[-1]
+  outcome <- rlang::sym(outcome)
+
+
+  r <- epi_recipe(epi_data) %>%
+    step_epi_ahead(!!outcome, ahead = args_list$ahead, skip = TRUE) %>%
+    recipes::update_role(!!outcome, new_role = "predictor") %>%
+    recipes::add_role(dplyr::all_of(keys), new_role = "predictor")
```

Comment:
I see these functions, `update_role()` and `add_role()` are also used where the flatline forecaster is demonstrated at the end of the getting started vignette... Since they seem quite useful, I think their use in preprocessing could be mentioned in a line where that is most appropriate to fit.

Reply:
In think @ChloeYou is doing a Vignette that talks about recipes/tidymodels more. Can you mention there?

Reply:
Noted! Will add to the vignette.